galaxy_ng/app/api/ui/serializers/execution_environment.py

```diff
@@ -218,29 +200,74 @@ def _content_info(self, content):
         }
 
         # TODO: Figure out if there is a way to prefetch Manifest and Tag objects
-        if content.pulp_type == 'container.manifest':
+        if content.pulp_type == "container.manifest":
             manifest = container_models.Manifest.objects.get(pk=content.pk)
-            return_data['manifest_digest'] = manifest.digest
-        elif content.pulp_type == 'container.tag':
-            tag = container_models.Tag.objects.select_related('tagged_manifest')\
-                .get(pk=content.pk)
-            return_data['manifest_digest'] = tag.tagged_manifest.digest
-            return_data['tag_name'] = tag.name
+            return_data["manifest_digest"] = manifest.digest
+        elif content.pulp_type == "container.tag":
+            tag = container_models.Tag.objects.select_related("tagged_manifest").get(pk=content.pk)
+            return_data["manifest_digest"] = tag.tagged_manifest.digest
+            return_data["tag_name"] = tag.name
 
         return return_data
 
 
 class ContainerReadmeSerializer(serializers.ModelSerializer):
-
     class Meta:
         model = models.ContainerDistroReadme
         fields = (
-            'updated',
-            'created',
-            'text',
+            "updated",
+            "created",
+            "text",
         )
 
         read_only_fields = (
-            'updated',
-            'created',
+            "updated",
+            "created",
         )
+
+
+class ContainerRegistryRemoteSerializer(
+    v3_serializers.LastSyncTaskMixin,
+    core_serializers.RemoteSerializer,
+):
+    created_at = serializers.DateTimeField(source='pulp_created', required=False)
+    updated_at = serializers.DateTimeField(source='pulp_last_updated', required=False)
+    last_sync_task = serializers.SerializerMethodField()
+    write_only_fields = serializers.SerializerMethodField()
+
+    class Meta:
+        model = models.ContainerRegistryRemote
+        fields = [
+            "pk",
+            "name",
+            "url",
+            "policy",
+            "created_at",
+            "updated_at",
+            "username",
+            "password",
+            "tls_validation",
+            "client_key",
+            "client_cert",
+            "ca_cert",
+            "last_sync_task",
+            "download_concurrency",
+            "proxy_url",
+            "proxy_username",
+            "proxy_password",
+            "write_only_fields",
+            "rate_limit",
+        ]
+        extra_kwargs = {
+            'name': {'read_only': True},
+            'client_key': {'write_only': True},
+        }
+
+    def get_write_only_fields(self, obj):
+        return utils.get_write_only_field(self, obj)
+
+    def get_last_sync_task_queryset(self, obj):
```

Comment:
the serializer method name should be `get_<field_name>` so `get_last_sync_task` to match `last_sync_task`

Reply:
Oh I think I see now... the `LastSyncTaskMixin` has `get_last_sync_task()` already defined, but needs the `get_last_sync_task_queryset()` implemented.

Reply:
@ShaiahWren I don't think any changes were needed here, but it looks like `get_last_sync_task_queryset()` was renamed to `get_last_sync_task()`. By using `get_last_sync_task_queryset()` we get a [standard set of data](https://github.com/ansible/galaxy_ng/blob/b483f2780d1ae21c6d6e922b0595b757758e0d12/galaxy_ng/app/api/v3/serializers/sync.py#L40-L46) rather than whatever it displays with just returning just an object in the `last_sync_task` field